substrate/frame/scheduler/src/lib.rs

```diff
@@ -1124,11 +1285,17 @@ impl<T: Config> Pallet<T> {
 			},
 			Err(()) => Err((Overweight, Some(task))),
 			Ok(result) => {
+				let failed = result.is_err();
```

Comment:
But still if you `take` retry here, it would be easier to maintain, if new branches appears below. Now you need to remove it in different branches and make an additional insert in `schedule_retry`. I would just pass the reference to `retry` from this context for `schedule_retry`.

Reply:
So, you mean if the task fails, check for a retry config and if necessary give that to `schedule_retry`? That sounds good.

Reply:
Done. PTAL